plugins/minigames/dodge_the_ball.py

```diff
@@ -325,33 +327,34 @@ def check_ball_type(self, ball_type: BallType) -> NoReturn:
             self.shoot_speed = 0.4
             # box light color and ball tex.
             self.light.color = (1.0, 0.0, 1.0)
-            self.ball_tex = ba.gettexture('egg3')
+            self.ball_tex = bs.gettexture('egg3')
         elif ball_type == BallType.HARD:
             self.ball_mag = 2500
             self.ball_gravity = 0.0
             # next ball shoot speed.
             self.shoot_speed = 0.6
             # box light color and ball tex.
             self.light.color = (1.0, 0.2, 1.0)
-            self.ball_tex = ba.gettexture('egg1')
+            self.ball_tex = bs.gettexture('egg1')
 
     def shoot_animation(self) -> NoReturn:
 
-        ba.animate(
+        bs.animate(
             self.node,
-            "model_scale", {
+            "mesh_scale", {
                 0.00: 1.4,
                 0.05: 1.7,
                 0.10: 1.4,
             }
         )
         # playing shoot sound.
-        ba.playsound(self.shoot_sound, position=self.node.position)
+        # self.shoot_sound, position = self.node.position.play();
+        self.shoot_sound.play()
```

Comment:
this won't take node's position into account.

i think `self.shoot_sound.play(position=self.node.position)` should fix it, will have to test tho.

Reply:
Surprisingly it worked!